test/lit/passes/precompute-gc.wast

```diff
@@ -34,4 +39,356 @@
+ ;; CHECK-NEXT:  (local $x (ref null $struct))
+ ;; CHECK-NEXT:  (local $y (ref null $struct))
+ ;; CHECK-NEXT:  (local $tempresult i32)
+ ;; CHECK-NEXT:  (local.set $x
+ ;; CHECK-NEXT:   (struct.new_with_rtt $struct
+ ;; CHECK-NEXT:    (i32.const 1)
+ ;; CHECK-NEXT:    (rtt.canon $struct)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (local.set $y
+ ;; CHECK-NEXT:   (local.get $x)
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (local.set $tempresult
+ ;; CHECK-NEXT:   (ref.eq
+ ;; CHECK-NEXT:    (local.get $x)
+ ;; CHECK-NEXT:    (local.get $y)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (local.get $tempresult)
+ ;; CHECK-NEXT: )
+ (func $new-ref-comparisons (result i32)
+  (local $x (ref null $struct))
+  (local $y (ref null $struct))
+  (local $tempresult i32)
+  (local.set $x
+   (struct.new_with_rtt $struct
+    (i32.const 1)
+    (rtt.canon $struct)
+   )
+  )
+  (local.set $y
+   (local.get $x)
+  )
+  ;; assign the result, so that propagate calculates the ref.eq
+  (local.set $tempresult
+   (ref.eq
+    (local.get $x)
+    (local.get $y)
+   )
+  )
+  ;; this value could be precomputed in principle, however, we currently do not
+  ;; precompute GC references, and so nothing will be done.
+  (local.get $tempresult)
+ )
+ ;; CHECK:      (func $propagate-equal (result i32)
+ ;; CHECK-NEXT:  (local $tempresult i32)
+ ;; CHECK-NEXT:  (local $tempref (ref null $empty))
+ ;; CHECK-NEXT:  (local.set $tempresult
+ ;; CHECK-NEXT:   (ref.eq
+ ;; CHECK-NEXT:    (local.tee $tempref
+ ;; CHECK-NEXT:     (struct.new_default_with_rtt $empty
+ ;; CHECK-NEXT:      (rtt.canon $empty)
+ ;; CHECK-NEXT:     )
+ ;; CHECK-NEXT:    )
+ ;; CHECK-NEXT:    (local.get $tempref)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (local.get $tempresult)
+ ;; CHECK-NEXT: )
+ (func $propagate-equal (result i32)
+  (local $tempresult i32)
+  (local $tempref (ref null $empty))
+  ;; assign the result, so that propagate calculates the ref.eq
+  (local.set $tempresult
+   (ref.eq
+    ;; allocate one struct
+    (local.tee $tempref
+     (struct.new_with_rtt $empty
+      (rtt.canon $empty)
+     )
+    )
+    (local.get $tempref)
+   )
+  )
+  ;; this value could be precomputed in principle, however, we currently do not
+  ;; precompute GC references, and so nothing will be done.
+  (local.get $tempresult)
+ )
+ ;; CHECK:      (func $propagate-unequal (result i32)
+ ;; CHECK-NEXT:  (local $tempresult i32)
+ ;; CHECK-NEXT:  (local $tempref (ref null $empty))
+ ;; CHECK-NEXT:  (local.set $tempresult
+ ;; CHECK-NEXT:   (ref.eq
+ ;; CHECK-NEXT:    (struct.new_default_with_rtt $empty
+ ;; CHECK-NEXT:     (rtt.canon $empty)
+ ;; CHECK-NEXT:    )
+ ;; CHECK-NEXT:    (struct.new_default_with_rtt $empty
+ ;; CHECK-NEXT:     (rtt.canon $empty)
+ ;; CHECK-NEXT:    )
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (local.get $tempresult)
+ ;; CHECK-NEXT: )
+ (func $propagate-unequal (result i32)
```

Comment:
Is this the case that the previous PR did not correctly handle?

Reply:
Yes, exactly.